faraday/proxy.py

```diff
@@ -19,21 +19,218 @@
+parser = argparse.ArgumentParser(description='Proxy application interfaces a Faraday radio over USB UART')
+parser.add_argument('--init-config', dest='init', action='store_true', help='Initialize Proxy configuration file')
+parser.add_argument('--callsign', help='Set Faraday callsign')
+parser.add_argument('--nodeid', type=int, help='Set Faraday node ID')
+parser.add_argument('--port', help='Set Faraday UART port')
+parser.add_argument('--baudrate', default='115200', help='Set Faraday UART baudrate')
+parser.add_argument('--timeout', type=int, default=5, help='Set Faraday UART timeout')
+parser.add_argument('--unit', type=int, default=0, help='Specify Faraday unit to configure')
+
+#Proxy options
+parser.add_argument('--number', type=int, default=0, help='Set number of Faraday radios to use')
+parser.add_argument('--log', action='store_true', help='Set Proxy into logging mode')
+parser.add_argument('--test-mode', dest='testmode', action='store_true', help='Set Proxy into test mode')
+parser.add_argument('--test-callsign', dest='testcallsign', help='Set Faraday test mode callsign')
+parser.add_argument('--test-nodeid', dest='testnodeid', type=int, help='Set Faraday test mode nodeid')
+parser.add_argument('--test-rate', dest='testrate', default=1, type=int, help='Set Faraday test mode rate')
+
+# Proxy database options
+parser.add_argument('--database', help='Set Faraday Proxy database')
+parser.add_argument('--schema', help='Set Faraday database schema')
+parser.add_argument('--test-database', dest='testdatabase', help='Set Faraday test mode database')
+parser.add_argument('--init-log', dest='initlog', action='store_true', help='Initialize Proxy log database')
+parser.add_argument('--save-log', dest='savelog', help='Save Proxy log database into new SAVELOG file')
+parser.add_argument('--showlogs', action='store_true', help='Show Proxy log database files')
+
+# Proxy Flask options
+parser.add_argument('--flask-host', dest='flaskhost', help='Set Faraday Flask server host address')
+parser.add_argument('--flask-port', type=int, dest='flaskport', help='Set Faraday Flask server port')
+
+# Parse the arguments
+args = parser.parse_args()
+
+
+def initializeProxyConfig():
+    '''
+    Initialize proxy configuration file from proxy.sample.ini
+
+    :return: None, exits program
+    '''
+
+    logger.info("Initializing Proxy")
+    shutil.copy(os.path.join(path, "proxy.sample.ini"), os.path.join(path, "proxy.ini"))
+    logger.info("Initialization complete")
+    sys.exit(0)
+
+
+def initializeProxyLog(config):
+    '''
+    Initialize the log database by deleting file
+
+    :param config: Proxy ConfigParser object from proxy.ini
+    :return: None
+    '''
+
+    logger.info("Initializing Proxy Log File")
+    log = config.get("DATABASE", "filename")
+    logpath = os.path.join(os.path.expanduser('~'), '.faraday', 'lib', log)
+    os.remove(logpath)
+    logger.info("Log initialization complete")
+
+
+def saveProxyLog(name, config):
```

Comment:
Not applicable, folder created if not already existing whenever proxy is run:

https://github.com/FaradayRF/Faraday-Software/pull/203/files#diff-a9d2cba55827323291e9d0c49fd6f720R629